packages/openid4vc-holder/src/presentations/OpenId4VpHolderService.ts

```diff
@@ -0,0 +1,236 @@
+
+    const builder = OP.builder()
+      .withResponseMode(ResponseMode.POST)
+      .withSupportedVersions([SupportedVersion.SIOPv2_ID1])
+      .withExpiresIn(300)
+      .withCheckLinkedDomain(CheckLinkedDomain.NEVER)
+      .withCustomResolver(this.getResolver(agentContext))
+
+    // Add did methods
+    for (const supportedDidMethod of supportedDidMethods) {
+      builder.addDidMethod(supportedDidMethod)
+    }
+
+    const op = builder.build()
+
+    return op
+  }
+
+  public async selectCredentialForProofRequest(
+    agentContext: AgentContext,
+    options: {
+      authorizationRequest: string | URI
+    }
+  ) {
+    const op = this.getOp(agentContext)
+
+    const verification = {
+      mode: VerificationMode.EXTERNAL,
+      resolveOpts: {
+        resolver: this.getResolver(agentContext),
+        noUniversalResolverFallback: true,
+      },
+    } satisfies Verification
+
+    // FIXME: this uses did-jwt for verification of the JWT, we can't verify it ourselves.
+    const verifiedAuthorizationRequest = await op.verifyAuthorizationRequest(options.authorizationRequest, {
+      verification,
+    })
+
+    if (!isVerifiedAuthorizationRequestWithPresentationDefinition(verifiedAuthorizationRequest)) {
+      throw new AriesFrameworkError(
+        'Only SIOPv2 authorization request including a single presentation definition are supported'
+      )
+    }
+
+    const selectResults = await this.presentationExchangeService.selectCredentialsForRequest(
+      agentContext,
+      verifiedAuthorizationRequest.presentationDefinitions[0].definition
+    )
+
+    return {
+      verifiedAuthorizationRequest,
+      selectResults,
+    }
+  }
+
+  /**
+   * Send a SIOPv2 authentication response to the relying party including a verifiable
+   * presentation based on OpenID4VP.
+   */
+  public async shareProof(
+    agentContext: AgentContext,
+    options: {
+      verifiedAuthorizationRequest: VerifiedAuthorizationRequestWithPresentationDefinition
+      submission: PresentationSubmission
+      submissionEntryIndexes: number[]
+    }
+  ) {
+    const op = this.getOp(agentContext)
+
+    const credentialsForInputDescriptor: CredentialsForInputDescriptor = {}
+
+    options.submission.requirements
+      .flatMap((requirement) => requirement.submission)
+      .forEach((submission, index) => {
+        const verifiableCredential = submission.verifiableCredentials[
+          options.submissionEntryIndexes[index] as number
+        ] as W3cCredentialRecord
+
+        const inputDescriptor = credentialsForInputDescriptor[submission.inputDescriptorId]
+        if (!inputDescriptor) {
+          credentialsForInputDescriptor[submission.inputDescriptorId] = [verifiableCredential.credential]
+        } else {
+          inputDescriptor.push(verifiableCredential.credential)
+        }
+      })
+
+    const vps = await this.presentationExchangeService.createPresentation(agentContext, {
```

Comment:
Does `vps` mean verifiablePresentations?

Reply:
i distructured the object for readability. Seemed also to me like verifiable presentations, but it contains multiple objects